core/lib/zksync_core/src/temp_config_store.rs

```diff
@@ -0,0 +1,47 @@
+use zksync_config::{
+    configs::{
+        api::{HealthCheckConfig, MerkleTreeApiConfig, Web3JsonRpcConfig},
+        chain::{
+            CircuitBreakerConfig, MempoolConfig, NetworkConfig, OperationsManagerConfig,
+            StateKeeperConfig,
+        },
+        house_keeper::HouseKeeperConfig,
+        FriProofCompressorConfig, FriProverConfig, FriWitnessGeneratorConfig, PrometheusConfig,
+        ProofDataHandlerConfig, ProverGroupConfig, WitnessGeneratorConfig,
+    },
+    ApiConfig, ContractsConfig, DBConfig, ETHClientConfig, ETHSenderConfig, ETHWatchConfig,
+    FetcherConfig, GasAdjusterConfig, ObjectStoreConfig, ProverConfigs,
+};
+
+// TODO (QIT-22): This structure is going to be removed when components will be respnsible for their own configs.
+/// A temporary config store allowing to pass deserialized configs from `zksync_server` to `zksync_core`.
+/// All the configs are optional, since for some component combination it is not needed to pass all the configs.
+#[derive(Debug)]
+pub struct TempConfigStore {
```

Comment:
This function resembles former `ZkSyncConfig`, but it isn't as dangerous:
- It cannot be created locally, given that `zksync_env_config` is not in dependencies.
- It doesn't implement `Clone`.
- It's only passed to `init_components` by reference.

Given that this structure is temporary, I think it's enough to guarantee that it won't spread.